fedimint-client/src/lib.rs

```diff
@@ -998,6 +998,12 @@ impl Client {
         operation_id: OperationId,
         mut partial_transaction: TransactionBuilder,
     ) -> anyhow::Result<(Transaction, Vec<DynState>, Range<u64>)> {
+        partial_transaction.inputs.extend(
+            self.primary_module()
+                .consolidate_inputs(self.primary_module_instance, dbtx, operation_id)
+                .await?,
+        );
+
```

Comment:
Good call to add these before `create_sufficient_input`, but imo we could do the same inside `create_sufficient_input`: first select consolidation inputs and then select the remaining amount (if any) using the normal algorithm.